extension/js/common/api/key-server/key-manager.ts

```diff
@@ -30,7 +31,12 @@ export class KeyManager extends Api {
   };
 
   private request = async <RT>(method: ReqMethod, path: string, vals?: Dict<any> | undefined, idToken?: string): Promise<RT> => {
-    return await Api.apiCall(this.url, path, vals, vals ? 'JSON' : undefined, undefined, idToken ? { Authorization: `Bearer ${idToken}` } : undefined, undefined, method);
-  };
-
-}
+    let res;
+    try {
+      res = await Api.apiCall(this.url, path, vals, vals ? 'JSON' : undefined, undefined, idToken ? { Authorization: `Bearer ${idToken}` } : undefined, undefined, method);
+    } catch (e) {
+      await Ui.modal.error(`Error while communicating with the key manager: ${e}`);
+    }
+    return res as RT;
+  }
```

Comment:
This is a wrong place to do handling of errors. The approach in handling errors in low-level method (like API method) is that they should be allowed to throw upwards.

Then upwards again and again, until they finally reach a place where we deal with UI (some `View`). Somewhere there, some user action has triggered the chain, and that's where we should be catching the error, once it already bubbled up.

Don't catch errors in low level methods this way. If we did that, you would have to write this code 100 times, for each low level method there is, which doesn't make sense.

Reply:
I tried to rewrite the `getPrivateKeys()` to return both keys and error then process the returned data where the function call has been made.

Reply:
why? either return result, or have it throw error. catch the error on the highest-level method that initiated the chain of calls.